api-service/database/licenses_test.go

```diff
@@ -0,0 +1,338 @@
+		mode := "summary"
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := "Italy"
+		environment := ""
+		olderThan := utils.MAX_TIME
+		out, err := m.db.SearchLicenses(mode, sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+
+		m.Require().NoError(err)
+		var expectedOut []interface{} = []interface{}{
+			map[string]interface{}{"_id": "Oracle ENT", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0},
+			map[string]interface{}{"_id": "Diagnostics Pack", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0},
+			map[string]interface{}{"_id": "Real Application Clusters", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0}}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+
+	m.T().Run("should_filter_out_by_environment", func(t *testing.T) {
+		mode := "summary"
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := ""
+		environment := "TEST"
+		olderThan := utils.MAX_TIME
+		out, err := m.db.SearchLicenses(mode, sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+
+		m.Require().NoError(err)
+		var expectedOut []interface{} = []interface{}{
+			map[string]interface{}{"_id": "Oracle ENT", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0},
+			map[string]interface{}{"_id": "Diagnostics Pack", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0},
+			map[string]interface{}{"_id": "Real Application Clusters", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0}}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+
+	m.T().Run("should_filter_out_by_older_than", func(t *testing.T) {
+		mode := "summary"
+		sortBy := ""
+		sortDesc := false
+		page := -1
+		pageSize := -1
+		location := ""
+		environment := ""
+		olderThan := utils.MIN_TIME
+		out, err := m.db.SearchLicenses(mode, sortBy, sortDesc, page, pageSize, location, environment, olderThan)
+
+		m.Require().NoError(err)
+		var expectedOut []interface{} = []interface{}{
+			map[string]interface{}{"_id": "Oracle ENT", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0},
+			map[string]interface{}{"_id": "Diagnostics Pack", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0},
+			map[string]interface{}{"_id": "Real Application Clusters", "compliance": true, "costPerProcessor": 0, "count": 0, "paidCost": 0, "totalCost": 0, "unlimited": false, "used": 0}}
+
+		assert.JSONEq(t, utils.ToJSON(expectedOut), utils.ToJSON(out))
+	})
+}
```

Comment:
Why don't you have the subtest order like other tests?

Reply:
I added `should_sort` test, is it ok?

Reply:
I mean the order of the subtest. You have subtest should_filter_out_by_older_than is the last instead of being the first like other tests

Reply:
Fixed in latest!